### PR TITLE
Fixing release and obfuscated build issues for flutter 3.x.x

### PR DESCRIPTION
**Pull request by srmanc**
This PR fixes release and obfuscation issues that were happending since flutter 2.8.x.

Please note that also example package was modified in order to promote the usage of 
```
@pragma('vm:entry-point')
```
for every top level or static method in order to avoid obfuscation.

Library also failed to send List<Capability?> arguments through ports, likely because class names could not be resolved internally by dart engine. And since only basic types should be sent trough ports, List<dynamic> suits and works instead as well.

Please review and merge this ASAP as a lot of projects depend on this.

### CODE REVIEW

**Comment by acoutts** [comment]
Note: I've had the same bug on iOS since upgrading to 3.3.0:

```
default	13:10:44.755017+0100	Runner	[VERBOSE-2:shell.cc(93)] Dart Error: Dart_LookupLibrary: library 'package:flutter_isolate/flutter_isolate.dart' not found.
default	13:10:44.755047+0100	Runner	[VERBOSE-2:dart_isolate.cc(668)] Could not resolve main entrypoint function.
default	13:10:44.755074+0100	Runner	[VERBOSE-2:dart_isolate.cc(167)] Could not run the run main Dart entrypoint.
default	13:10:44.755234+0100	Runner	[VERBOSE-2:runtime_controller.cc(385)] Could not create root isolate.
default	13:10:44.755260+0100	Runner	[VERBOSE-2:shell.cc(604)] Could not launch engine with configuration.
```

**Comment by acoutts** [comment]
Update: my issue of it not appearing to work is because even though I did a dependency override for flutter_isolate in the isolate_handler package to use your PR, my app which consumes isolate_handler still had the pub version of flutter_isolate in the lock file.

I added the dependency override for flutter_isolate to my app pubspec as well, and just had to decorate the function sent to isolate_handler (which then sends it to flutter_isolate), and it's now working.

**Comment by srmanc** [comment]
Yes, decorations are mandatory.

Additional note: sending simple classes (containing basic types) through ports does not work anymore, it actually crashes the app, so serialization / deserialization with Map is needed.

**Comment by nmfisher** [comment]
Thanks @srmanc

**Comment by deadsoul44** [comment]
This is merged but there is no release. Could you please publish a new version on pub.dev?

**Comment by etozzato** [comment]
`hello everyone!`

For those that are using both `flutter_isolate` and `isolate_handler`, this is the fix to the problem until the package becomes available to pub:

**Comment by srmanc** [comment]
@nmfisher Can we do a release on pub.dev?

**Comment by nmfisher** [comment]
I’ll try and get to this tomorrow

File: example/lib/main.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@ import 'package:flutter_isolate/flutter_isolate.dart';
 import 'package:path_provider/path_provider.dart';
 import 'package:flutter_downloader/flutter_downloader.dart';
 
+@pragma('vm:entry-point')
```
Comment (mzdm) [inline example/lib/main.dart:8]: @nmfisher also please add that this `@pragma('vm:entry-point')` is needed to the changelog

Comment (nmfisher) [reply example/lib/main.dart:8]: Added!

**Comment by nmfisher** [comment]
v2.0.3 now available on pub.dev:

https://pub.dev/packages/flutter_isolate/versions/2.0.3

**Comment by srmanc** [comment]
Great, thanks!

**Pull request merged**